server/task/src/main/java/org/apache/james/task/TaskManager.java

```diff
@@ -58,6 +59,9 @@ public boolean isFinished() {
         }
     }
 
+    class ReachedTimeoutException extends Exception {
```

Comment:
why the hell a checked Exception?

Reply:
to force handling it

Reply:
usually we use a RuntimeException for these cases, and explicitly state that the method is throwing it with a `throws MyRuntimeException`. What's the point here to force all the callers to handle it?

Reply:
We forgot to handle it at the right places, it would be a shame to allow this kind of bug to happen again. It's a design fix.